File: Source/DotNET/Model/Interactions/EventInteractionTrigger.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Scene.Model.Interactions;

/// <summary>
/// Represents an interaction started by observing a modeled domain event.
/// </summary>
/// <param name="EventName">The event being observed.</param>
public record EventInteractionTrigger(string EventName) : InteractionTrigger;

File: Source/JavaScript/model/interactions/EventInteractionTrigger.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { InteractionTrigger } from './InteractionTrigger';

/**
 * An interaction started by observing a modeled domain event.
 */
export interface EventInteractionTrigger extends InteractionTrigger {
    eventName: string;
}

export const EventInteractionTriggerPropertyNames: (keyof EventInteractionTrigger)[] = ['eventName'];
